devicemodel/include/dm.h

```diff
@@ -33,10 +33,9 @@
 #include "types.h"
 #include "dm_string.h"
 
-#define MAX_VMNAME_LEN	128U
+#define MAX_VMNAME_LEN	16U
```

Comment:
The VM name length is now part of the public hypercall interface, and we'd better put this macro to the header that defines the structures. I assume there is a unique header that are shared by all ACRN components.

Unfortunately I don't have a quick answer to how we can guarantee the sync between the length in C and in XSD. As a part of the public hypercall interface, I don't think it will be changed frequently. In the worst case where a user gives a loooong string which exceeds this limit, a compiler error will be generated which captures the error early (but not that friendly). An example of the error reads like this:

```
test.c:1:16: warning: initializer-string for array of ‘char’ is too long
    1 | char buf[16] = "a loooooong string";
      |                ^~~~~~~~~~~~~~~~~~~~
```

Reply:
It is not something we'll want to change very often if it's part of the public hypercall interface. I personnally felt that 15 characters were not an awful lot, the `ACRN_Service_VM` string we use already consumes all of these. Should be define this as a longer string to be on the safe side, I don't believe there would be much, if any, downside to a longer string?

Reply:
15 char restriction is to reuse the uuid field in kernel hypercall interface and we have no choice. The interface changing is not a technical issue but it will cause other compatible issue since the struct has been upstream already, and get the new change upstream is really a tough experience. 
If you think 15 char string as a VM name identifier is not comprehensive maybe we can add a prefix for specific VM, like "ACRN_Service_VM_"+ (15char_identifier:xxxx), "ACRN_Postlaunched_VM_"+ (15char_identifier:xxxx), but this is just my idea.

Reply:
Nah, I don't think it's worth the effort. It is a bit of a constraint but nothing really that would block the user, I don't think it will block anyone. Let's stick to the 15 characters limit and document it properly.

Reply:
@junjiemao1 Why are there two different #DEFINEs with slightly different names: MAX_VMNAME_LEN and MAX_VM_NAME_LEN?  Shouldn't there be only one?

Reply:
will use MAX_VM_NAME_LEN in next version